R/utils.R

```diff
@@ -0,0 +1,117 @@
+library(shiny)
+library(shinyGovstyle)
+library(RCurl)
+library(stringr)
+
+#' Custom disconnect message
+#'
+#'
+#' @description
+#' Create the html overlay panel to appear when RSConnect disconnects
+#'
+#' @param refresh
+#' @param links A list of mirros or alternative links to the dashboard
+#' @param publication_name The parent publication name
+#' @param publication_link The link to the publication on Explore Education Statistics
+#'
+#' @return
+#' @export
+#'
+#' @examples
+customDisconnectMessage <- function(refresh = "Refresh page",
+                                    links = sites_list,
+                                    publication_name = ees_pub_name,
+                                    publication_link = ees_publication) {
+  # Check links are valid
+
+  is_valid_sites_list <- function(sites){
+    lapply(str_trim(sites), startsWith, "https://department-for-education.shinyapps.io/")
+  }
+
+  if(FALSE %in% is_valid_sites_list(links)) {
+    stop("You have entered an invalid site link in the sites_list argument.")
+  }
+
+  is_valid_publication_link <- function(link){
+    startswith <- startsWith(str_trim(link), c("https://explore-education-statistics.service.gov.uk/find-statistics/",
+                                     "https://www.explore-education-statistics.service.gov.uk/find-statistics/",
+                                     "https://www.gov.uk/",
+                                     "https://gov.uk/"))
+  }
+
+  if(url.exists(publication_link) == FALSE | (TRUE %in% is_valid_publication_link(publication_link)) == FALSE) {
+    stop("You have entered an invalid publication link in the ees_publication argument.")
```

Comment:
Looks like the argument is publication_link rather than ees_publication.

Reply:
Nah think this is right bc it's pulling from here https://github.com/dfe-analytical-services/shiny-template/blob/47a5de81b64231e4282da41c2440c73c7def021c/global.R#L90C1-L91C1

Reply:
Hmm, but publication_link is the name of the variable as this function knows it and someone could run it by explicitly passing it a string (or some other named string variable) rather than the giving it ees_publication as defined in the template.